Concurrency problem with Grib2 files using netcdfAll-5.4.1

NullPointerException when opening two grib2 files concurrently.

version: netcdfAll-5.4.1
java: openjdk version "1.8.0_292" (Ubuntu), java version "11.0.9" 2020-10-20 LTS (Windows)
OS: ubuntu, windows

```
java.io.IOException: java.lang.NullPointerException
        at ucar.nc2.NetcdfFiles.open(NetcdfFiles.java:279)
        at ucar.nc2.dataset.NetcdfDatasets.openProtocolOrFile(NetcdfDatasets.java:455)
        at ucar.nc2.dataset.NetcdfDatasets.openFile(NetcdfDatasets.java:347)
        at ucar.nc2.dataset.NetcdfDatasets.openFile(NetcdfDatasets.java:320)
        at Test.run(Test.java:29)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
        at ucar.nc2.grib.grib2.table.WmoCodeFlagTables.getCodeTable(WmoCodeFlagTables.java:83)
        at ucar.nc2.grib.grib2.table.Grib2Tables.getCodeTableValue(Grib2Tables.java:209)
        at ucar.nc2.grib.grib2.table.NcepLocalTables.getCodeTableValue(NcepLocalTables.java:185)
        at ucar.nc2.grib.collection.Grib2Iosp.makeVariableLongName(Grib2Iosp.java:148)
        at ucar.nc2.grib.collection.Grib2Iosp.makeVariableLongName(Grib2Iosp.java:164)
        at ucar.nc2.grib.collection.GribIospBuilder.makeGroup(GribIospBuilder.java:298)
        at ucar.nc2.grib.collection.GribIospBuilder.addGroup(GribIospBuilder.java:75)
        at ucar.nc2.grib.collection.GribIosp.build(GribIosp.java:172)
        at ucar.nc2.NetcdfFiles.build(NetcdfFiles.java:784)
        at ucar.nc2.NetcdfFiles.open(NetcdfFiles.java:723)
        at ucar.nc2.NetcdfFiles.open(NetcdfFiles.java:276)
        ... 5 more
```

Steps to reproduce:
- download two small grib2 files from NOAA: `https://nomads.ncep.noaa.gov/pub/data/nccf/com/gfs/prod/gfs.20210610/00/atmos/gfs.t00z.goessimpgrb2.0p25.f000` and `https://nomads.ncep.noaa.gov/pub/data/nccf/com/gfs/prod/gfs.20210610/00/atmos/gfs.t00z.goessimpgrb2.0p25.f003` (you need to change to more recent date in the URL later)
- compile and run the below java code with `netcdfAll-5.4.1.jar` in the classpath and the downloaded NOAA files in the same directory

```
import java.io.IOException;

import ucar.nc2.NetcdfFile;
import ucar.nc2.Variable;
import ucar.nc2.dataset.NetcdfDatasets;

public class Test implements Runnable {

  private static String testFile1 = "gfs.t00z.goessimpgrb2.0p25.f000";
  private static String testFile2 = "gfs.t00z.goessimpgrb2.0p25.f003";

  public static void main(String[] args) {

    Thread t1 = new Thread(new Test(testFile1));
    Thread t2 = new Thread(new Test(testFile2));

    t1.start();
    t2.start();
  }

  private String filename;

  public Test(String filename) {
    this.filename = filename;
  }

  @Override
  public void run() {
    try (NetcdfFile ncfile = NetcdfDatasets.openFile(filename, null)) {
      Variable var = ncfile.findVariable("time");
      System.out.println(filename + " open, 'time' size is " + var.getSize());
    } catch (IOException e) {
      e.printStackTrace();
    }
  }

}
```

--- a/grib/src/main/java/ucar/nc2/grib/grib2/table/WmoCodeFlagTables.java
+++ b/grib/src/main/java/ucar/nc2/grib/grib2/table/WmoCodeFlagTables.java
@@ -47,7 +47,7 @@ public class WmoCodeFlagTables {
 
   private static WmoCodeFlagTables instance;
 
-  public static WmoCodeFlagTables getInstance() {
+  public static synchronized WmoCodeFlagTables getInstance() {
     if (instance == null) {
       instance = new WmoCodeFlagTables();
       try {

